src/templates.rs

```diff
@@ -1734,11 +1734,22 @@ pub(crate) fn option_type(inner: &Schema, gen_state: &GenState) -> Result<String
             format!("Option<{}>", nested_type)
         }
 
-        Schema::Record(RecordSchema {
-            name: Name { name, .. },
-            ..
-        })
-        | Schema::Enum(EnumSchema {
+        Schema::Record(rec) => {
+            let recursive = rec.fields.iter().any(|f| {
+                matches!(&f.schema, Schema::Union(scm) if scm.variants().iter().any(|v|
+                    matches!(v, Schema::Ref { name } if name == &rec.name)
+                ))
+            });
```

Comment:
Isn't this checking only one level of recursion and not a complete recursion ?

Reply:
you're right, I've changed it so that nested recursions are also detected.

Reply:
awesome !